packages/realm-react/src/AppProvider.tsx

```diff
@@ -18,11 +18,43 @@
 import { isEqual } from "lodash";
 import React, { createContext, useContext, useLayoutEffect, useRef, useState } from "react";
 import Realm from "realm";
+import { AuthOperationName, AuthResult, OperationState } from "./types";
+
+type AppContextValue = {
+  app: Realm.App | null;
+};
 
 /**
  * Create a context containing the Realm app.  Should be accessed with the useApp hook.
  */
-const AppContext = createContext<Realm.App | null>(null);
+const AppContext = createContext<AppContextValue>({
+  app: null,
+});
+
+type AppOpStateValue = {
+  authOperationStateHook: [AuthResult, React.Dispatch<React.SetStateAction<AuthResult>>] | null;
+};
+
+const AuthOpStateContext = createContext<AppOpStateValue>({
+  authOperationStateHook: null,
+});
+
+type AuthOpStateProviderProps = { children: React.ReactNode };
+
+const AuthOpStateProvider: React.FC<AuthOpStateProviderProps> = ({ children }) => {
+  const [authOpResult, setAuthOpResult] = useState<AuthResult>({
+    state: OperationState.NotStarted,
+    pending: false,
+    success: false,
+    error: undefined,
+    currentOperation: AuthOperationName.None,
```

Comment:
Since there are only supposed to be a single operation in flight at any given time, I think `current` is redundant.

```suggestion
    operation: AuthOperationName.None,
```